dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -1401,13 +1437,9 @@ def dpnp_pinv(a, rcond=1e-15, hermitian=False):
 
     """
 
-    if a.size == 0:
+    if _is_empty_2d(a):
         m, n = a.shape[-2:]
-        if m == 0 or n == 0:
-            res_type = a.dtype
-        else:
-            res_type = _common_type(a)
-        return dpnp.empty_like(a, shape=(a.shape[:-2] + (n, m)), dtype=res_type)
+        return dpnp.empty_like(a, shape=(a.shape[:-2] + (n, m)), dtype=a.dtype)
```

Comment:
no need to pass dtype, because it will be inherit from `a` due to `_like` function used
```suggestion
        return dpnp.empty_like(a, shape=(a.shape[:-2] + (n, m)))
```